### PR TITLE
[PAR-4464] magento additional offer requests made during add to cart event

### PR DESCRIPTION
**Pull request by adityaagashe1**
# Description

This PR will address two bugs it is added in the tickets section. We were making additional offers call due to
1. Offers call was made for Extend PP LineItem in Cart. 
2. In minitcart js file Category is not available hence we decided not to send category for any offer surface. 

## Ticket
1. https://helloextend.atlassian.net/browse/PAR-4464
4. https://helloextend.atlassian.net/browse/PAR-4465

List your ticket here

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (new code changes but everything functions the same)

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have deployed and validated my code on a sandbox
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: view/frontend/web/js/view/cart/product-protection-simple-offer.js [hunk 1]
```diff
@@ -43,13 +43,16 @@ define(['cartUtils', 'extendSdk', 'ExtendMagento'], function (cartUtils, Extend,
...
   return function (config) {
+    const sku = config[0].selectedProductSku
     const activeProductData = {
       referenceId: config[0].selectedProductSku,
       price: config[0].selectedProductPrice * 100,
       category: config[0].productCategory,
       onAddToCart: handleAddToCartClick,
     }
     Extend.config({ storeId: config[0].extendStoreUuid, environment: config[0].activeEnvironment })
+    if (sku === 'extend-protection-plan') return
```
Comment (jay-faust-extend) [inline view/frontend/web/js/view/cart/product-protection-simple-offer.js]: Should this be before we call `Extend.config()`?

**Comment by adityaagashe1** [comment]
Resolved!

File: view/frontend/web/js/view/cart/product-protection-simple-offer.js (hunk 1 above)
Comment (adityaagashe1) [reply view/frontend/web/js/view/cart/product-protection-simple-offer.js]: Changed the location

File: view/frontend/web/js/view/cart/minicart-updates.js [hunk 2]
```diff
@@ -14,14 +14,14 @@ define(['jquery', 'cartUtils', 'extendSdk', 'ExtendMagento'], function (
   const itemDetailsSelector = 'div.product-item-details'
   const simpleOfferClass = 'extend-minicart-simple-offer'
 
-  const handleUpdate = function () {
+  const handleUpdate = function (config) {
```
Comment (jay-faust-extend) [inline view/frontend/web/js/view/cart/minicart-updates.js:17]: Looks like we don't use `config`

**Review by jay-faust-extend — APPROVED** [review]

File: ViewModel/CategoriesForMinicartProducts.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<?php
```
Comment (simsben) [inline ViewModel/CategoriesForMinicartProducts.php:1]: So you're adding this file and some of the other minicart changes for later when we need it?

**Review by simsben — APPROVED** [review]

**Pull request merged**